libs/sdk-ui-filters/src/AttributeFilterHandler/types/elementsLoader.ts

```diff
@@ -453,6 +453,17 @@ export interface IAttributeElementLoader {
      * @param callback - callback to run
      */
     onLoadCustomElementsCancel: CallbackRegistration<OnLoadCustomElementsCancelCallbackPayload>;
+
+    /**
+     * Sets information whether elements should be reinitialized for example
+     * in case when limiting attributes changed.
+     */
+    setShouldReloadElements(shouldReloadElements: boolean): void;
```

Comment:
Why do we need this?

Reply:
This seemed to me as the easiest way how to avoid immediate elements reload in all child filters after the parents changes (this is part of the requirement, to avoid the loading cascade because of circulars and to avoid many requests). Instead of immediate reload, we just set the flag for each affected filter and when we need it, we reload them later. In our case this is when we open the child filter, but I can imagine we could also use this to reload asynchronously when we know we can safely do it.

Reply:
Can we solve this rather by checking for the backend capability related to the circular dependencies (and if we need some state for it, move it to the react hook)?

It seems this does not need to be in the AttributeFilterHandler at all - it's just dummy setter/getter and in context of the handler, it does nothing.

Reply:
I don't see a way how to avoid using handler here. We need one filter to trigger some change in the state of the second filter and persist this for later usage in the second filter. Every state change is done using the handler and there are many dummy getters and setters (almost for each item in the state). Am I missing something here?

Reply:
We need the flag in the state in order to distinguish correct actions when filter is opened (reload elements when filters changed). We use both setter and getter because information whether dropdown is open is not in the state of handler but only in the local state of the dropdown component. Would it help to rename the flag to something like `limitingFiltersChanged`?

We can set the flag when limiting filters are being changed in the state (redux action `setLimitingAttributeFilters`), but the getter and setter still have to be exposed because we have to invalidate the flag when filter is opened.

Reply:
I'm not contradicting your points, that you probably need some state to store this information.

I'm contradicting where this state should be stored -> definitely not in the handler itself (it's not affecting the handler in any way), we should likely rather put in the react hook local state).

Reply:
Moved to a separate hook with local state. Thank you